### PR TITLE
fix: solution of issue2268 draft

### LINKED ISSUE #2126: [Bug] [Gitlab Plugin] Gitlab users are not appeared in table.users
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar issues.

### What happened

I set up DevLake v0.11-test version and imported Gitlab and Jira data. 
When I tried to get Gitlab issue creator's detail by joining table.users. I didn't get the full user info, such as organization. Thus, I couldn't tell if an issue is created by a community member.

### What you expected to happen

GitLab users appeared in table.users.

### How to reproduce

1. Set up DevLake main
2. Collected Gitlab data
3. Connected to MySQL
4. Run query
`select distinct i.creator_id, i.creator_name, u.id, u.name FROM issues i
left join users u on i.creator_id = u.id`

### Anything else

It's related to #2122

### Version

main

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### LINKED ISSUE #2268: [GitLab] table.issues and pull_requests lack creator_id, author_id, etc.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar issues.

### What happened

After the collection of GitLab data, the domain layer seemed not so 

- [x] table.issues lacks the values of `creator_id`, `creator_name`, and `assignee_id` resolved by #2396 
- [ ] table.pull_requests lacks the values of `author_id`
- [ ] mr_notes in the tool layer doesn't show in the domain layer table pull_request_comments

### What you expected to happen

- [ ] `creator_id`, `creator_name`, and `assignee_id` have values in table.issues. `creator_id` is the FK to table.accounts  (table.users before v0.12)
- [ ] `author_id` has value, and is the FK  to table.accounts (table.users before v0.12)
- [ ] mr_notes from GitLab appeared in table.pull_request_comments
- [ ] add a column `type` in domain layer's table.pull_request_comments, please see the value assignment logic in the section 'Definition of GitLab comments'.

### How to reproduce

1. setup DevLake v0.11
2. collect any GitLab repo

### Definition of GitLab comments
1. Gitlab consists of 2 types of comments:
   a. normal comments: type = null, system = false 
     image
   b. diff comments: type = diffNote, system = false
     image

2. For values of the `type` field in table.pull_request_comments
   a. for normal comments: if it can't be set to null then empty string
   b. for diff comments: DIFF
   c. for review comments (exist in GitHub but not GitLab, see pic below): REVIEW
     image

### Anything else

related to #2126

### Version

main

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by warren830**
# Summary

creator_id, creator_name, and assignee_id, author_id have values in table,
creator_id and author_id are the FK to table.users(after v0.12 is table.account)
add a column type in domain layer's table.pull_request_comments

### Does this close any open issues?
closes #2268 
closes #2126 

### Screenshots
Include any relevant screenshots here.

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: plugins/github/tasks/commit_convertor.go [hunk 1]
```diff
@@ -79,14 +78,14 @@ func ConvertCommits(taskCtx core.SubTaskContext) error {
...
-				AuthorId:       userDidGen.Generate(data.Options.ConnectionId, githubCommit.AuthorId),
+				AuthorId:       githubCommit.AuthorEmail,
 				AuthorName:     githubCommit.AuthorName,
 				AuthorEmail:    githubCommit.AuthorEmail,
 				AuthoredDate:   githubCommit.AuthoredDate,
 				CommitterName:  githubCommit.CommitterName,
 				CommitterEmail: githubCommit.CommitterEmail,
 				CommittedDate:  githubCommit.CommittedDate,
-				CommitterId:    userDidGen.Generate(data.Options.ConnectionId, githubCommit.CommitterId),
+				CommitterId:    githubCommit.CommitterEmail,
```
Comment (abeizn) [inline plugins/github/tasks/commit_convertor.go:88]: CommitterId and AuthorId use email?

File: plugins/gitlab/tasks/commit_convertor.go [hunk 2]
```diff
@@ -74,14 +74,14 @@ func ConvertApiCommits(taskCtx core.SubTaskContext) error {
...
-			commit.AuthorId = userDidGen.Generate(data.Options.ConnectionId, gitlabCommit.AuthorName)
+			commit.AuthorId = gitlabCommit.AuthorEmail
 			commit.AuthorName = gitlabCommit.AuthorName
 			commit.AuthorEmail = gitlabCommit.AuthorEmail
 			commit.AuthoredDate = gitlabCommit.AuthoredDate
 			commit.CommitterName = gitlabCommit.CommitterName
 			commit.CommitterEmail = gitlabCommit.CommitterEmail
 			commit.CommittedDate = gitlabCommit.CommittedDate
-			commit.CommitterId = userDidGen.Generate(data.Options.ConnectionId, gitlabCommit.AuthorName)
+			commit.CommitterId = gitlabCommit.CommitterEmail
```
Comment (abeizn) [inline plugins/gitlab/tasks/commit_convertor.go:84]: same problem

File: plugins/github/tasks/commit_convertor.go (hunk 1 above)
Comment (warren830) [reply plugins/github/tasks/commit_convertor.go:88]: > CommitterId and AuthorId use email?

Yes, the same to gitextractor

**Review by abeizn — APPROVED** [review]
LGTM

**Pull request merged**